Improve log-in analytics

xsyann opened the pull request:
Rename `Sign In` event to `Log In` and throws it even when the user resume log-in.

The property `type` (`password` or `resume`) allows to track both types of log-in.

Donorhan reviewed (approved):
Thanks, it's ok for me! I'm waiting for a review from @RobinLbt

RobinLbt commented:
Hey @xsyann, why do you want to rename the event?

It's a bit annoying on our side since we'll have to join the old event `Sign In` with the new `Log In` every time we want to make an analysis with all the historical data we already tracked

As for the `type` property, it makes sense but most analytics tools have default session events

For instance, we use Mixpanel and we have those: 
<img width="634" alt="image" src="https://user-images.githubusercontent.com/25883677/185875451-1245be77-49c1-4ec7-bada-0631088ebe32.png">

Which analytics tool are you using?
Maybe they have it too. 
If not, then yes we can add the property :)

xsyann commented:
Hi @RobinLbt,

I just wanted to rename the event for two reasons:

1. When I read `Sign In`, I think more about a form input signing in, and as I added the resume login event, I thought that `Log In` was more accurate than `Sign In`
2. To keep a consistency with the `Log out` event that I wanted to add

Nonetheless, if it induces a complexity on your side, I'm perfectly fine with 'Sign In', it doesn't really matter :)

We are also using Mixpanel, I guess you are using timeout based session then?
I thought that it was not very expensive to configure event based session if we added exhaustive login events (password + resume) and logout events:

<img width="529" alt="Screenshot 2022-08-29 at 08 37 02" src="https://user-images.githubusercontent.com/1092138/187140299-85771305-8f6c-4f09-9a4a-7ef936b18bee.png">

RobinLbt commented:
Yes you are right, we are using a timeout-based session which is not ideal to be honest 

The event-based solution seems better

I do not see the new `Log Out` event in your commit, is it intentional?

xsyann commented:
I've done it in a separated branch, I've just created the PR: https://github.com/l3mpire/lemverse/pull/85

Donorhan commented:
@RobinLbt Is everything good for you?
